utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImpl.kt

```diff
@@ -35,34 +47,24 @@ class PerformanceMetricsAssessorImpl @Inject constructor(
 
   override fun getTotalPssUsed(): Long {
     val pid = ActivityManager.RunningAppProcessInfo().pid
-    activityManager.runningAppProcesses
     val processMemoryInfo = activityManager.getProcessMemoryInfo(intArrayOf(pid))
     return processMemoryInfo?.map { it.totalPss }?.sum()?.toLong() ?: 0L
   }
 
-  override fun getDeviceStorageTier(): OppiaMetricLog.StorageTier {
-    val permanentStorageUsage = context.filesDir?.listFiles()?.map { it.length() }?.sum()
-    val cacheStorageUsage = context.cacheDir?.listFiles()?.map { it.length() }?.sum()
-    val usedStorage = (cacheStorageUsage?.let { permanentStorageUsage?.plus(it) })
-    if (usedStorage != null) {
-      return when (
-        usedStorage.let { it / (1024 * 1024 * 1024) }
-      ) {
-        in 0L until 32L -> OppiaMetricLog.StorageTier.LOW_STORAGE
-        in 32L..64L -> OppiaMetricLog.StorageTier.MEDIUM_STORAGE
-        else -> OppiaMetricLog.StorageTier.HIGH_STORAGE
-      }
+  override fun getDeviceStorageTier(): OppiaMetricLog.StorageTier =
+    when (getUsedStorage()) {
+      in 0L until lowStorageTierUpperBound -> LOW_STORAGE
```

Comment:
The range logic still confuses me. Wouldn't we want the tier to be LOW_STORAGE if used storage == lowStorageTierUpperBound?

FWIW you could redo the logic here to make it a bit easier to do deal with edge cases:

```kotlin
  val usedStorage = getUsedStorage()
  return when {
    usedStorage <= lowStorageUpperBound -> LOW_STORAGE
    usedStorage <= mediumStorageTierUpperBound -> MEDIUM_STORAGE
    else -> HIGH_STORAGE
  }
```

Ditto for memory tiers below.

Reply:
done.